tools/tscdocgen/main.go

```diff
@@ -1115,6 +1115,13 @@ func (e *emitter) createCodeDetails(node *typeDocNode) string {
 	}
 }
 
+// @pulumi/awsx imports types within the same modules using the following.
+// We'll use this mapping to map the local import to the actual top-level exported module.
+var awsxImportModuleMap = map[string]string{
+	"ecs.": "ecs",
+	"mod.": "lb",
```

Comment:
i'm not sure ths is always valid.  awsx may import the same module it is in using 'mod' in many different places.  Note: if there's a module import pattern we should be using instead, lmk and we can update awsx to consistently follow it .

Reply:
> i'm not sure ths is always valid. awsx may import the same module it is in using 'mod' in many different places. 

I checked awsx and it's currently only using `import * as mod from ".";` in `lb`.

> Note: if there's a module import pattern we should be using instead, lmk and we can update awsx to consistently follow it .

@CyrusNajmabadi, what do you think about importing the root, and using the same prototypical naming we'd use as a consumer of the library, when referring to public exported types within the library?

For example, instead of:

```typescript
import * as mod from ".";
import * as x from "..";
```

And referring to types using `mod.Foo` and `x.mod.Foo`, do the following instead:

```typescript
import * as awsx from "..";
```

And then always refer to public export types using the full `awsx.mod.Foo` name?

It does mean we'd have to be slightly more verbose in the implementation of the library when referring to such types, but since TypeDoc only provides the name of the type as written, this would allow us to easily do linking of types within a library without any special casing in the doc generator other than seeing the type name starts with `awsx.`.